### PR TITLE
feat: Add support for Typer

### PR DESCRIPTION
**Pull request by evanmags**
Hey @willmcgugan, You don't have a CONTRIBUTING.md on this repo, so I'm not sure if/how I can contribute to this, but these are the changes I made on my fork to get Trogon working in my Typer app. I have a few TODOs below but would like your input for desired usage and best practice. Thank you 🙏 

TODO:
- [ ] match desired usage 
- [ ] tests based on best practice

### CODE REVIEW

File: trogon/trogon.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
     Footer,
 )
 from textual.widgets.tree import TreeNode
+import typer
```
Comment (willmcgugan) [inline trogon/trogon.py:26]: We don't want to force a dependancy on typer, for those who just want to use click...

**Comment by evanmags** [comment]
Removed the dependency requirement on typer and created an extra for it. This pattern of `init_tui(app)` feels more intuitive than shoehorning the decorator to work with a library that suggests creating many nested apps.

I'm weighing the benefits of being able to add this to a single command in typer, as the pattern set out in the docs would be to create a nested app for that.

File: README.md [hunk 2]
```diff
@@ -100,6 +101,15 @@ pip install trogon
...
 3. Your click app will have a new `tui` command available.
 
+### Typer
+1. Import `from trogon.typer import init_tui`
+2. Add the `tui` decorator above your typer app. e.g.
+    ```python
+    cli = typer.Typer(...)
+    init_tui(cli)
+    ```
+3. Your click app will have a new `tui` command available.
```
Comment (darrenburns) [inline README.md:111]: Think this needs some work - it still talks about Click and decorators etc.

**Review by darrenburns — DISMISSED** [review]

**Comment by mgielda** [comment]
Hi @darrenburns - I made a PR against @evanmags' branch fixing the docs. When he merges that, your review will be included is there anything else that needs to be done to make this land?

**Review by daneah — APPROVED** [review]

**Pull request merged**